packages/RecentChangesAPI/src/custom/modules/Users/clients/base/api/CustomRecentChangesApi.php

```diff
@@ -0,0 +1,214 @@
+     * @return array Array of query results. Each module in $modules will have its own set of results in the array.
+     */
+    protected function queryForRecentChanges($dateTimeOne, $dateTimeTwo, $modules, $addDeleted)
+    {
+        $arrayOfQueryResults = array();
+
+        foreach ($modules as $module) {
+            $query = new SugarQuery();
+            $query->distinct(true);
+            $query->from(BeanFactory::newBean($module), array('team_security' => true, 'add_deleted' => $addDeleted));
+            //TODO:  I was getting invalid link errors for modules like Accounts and Opportunities when I tried to do a
+            // join instead of a joinTable.  Is there a better way to get the table name than lower casing the module name?
+            $tableName = strtolower($module);
+            $query->joinTable('users')->on()
+                ->equalsField("$tableName.assigned_user_id", 'users.id');
+            $query->select(array('users.id', 'users.user_name'));
+            $query->where()->gt('date_modified', $dateTimeOne);
+            $query->where()->lte('date_modified', $dateTimeTwo);
+            $result = $query->execute();
+            $arrayOfQueryResults[] = $result;
+        }
+        return $arrayOfQueryResults;
+    }
+
+    /**
+     * Ensures the since param has been passed and that it uses the correct date format
+     * @param $sinceParam The param passed in as an argument
+     * @return mixed The validated param
+     * @throws SugarApiExceptionInvalidParameter if the since param is not in the correct date format
+     * @throws SugarApiExceptionMissingParameter if the since param is not passed
+     */
+    protected function validateSinceParam($sinceParam){
+        if (empty($sinceParam)) {
+            throw new SugarApiExceptionMissingParameter('Missing required parameter: since');
+        }
+
+        $errors = $this->validator->validate($sinceParam, new Assert\DateTime(array('format' => $this->dateFormat)));
+        if (count($errors) > 0) {
+            throw new SugarApiExceptionInvalidParameter((string)$errors . '    since param should be of the format '
+                . $this->dateFormat . '.');
+        }
+        return $sinceParam;
+    }
+
+    /**
+     * Ensures the add_deleted param has been passed and that it is a boolean
+     * @param $AddDeletedParam The param passed in as an argument
+     * @return mixed The validated param
+     * @throws SugarApiExceptionInvalidParameter if the param is not a boolean
+     * @throws SugarApiExceptionMissingParameter if the param is not passed
+     */
+    protected function validateAddDeletedParam($AddDeletedParam){
+        if (empty($AddDeletedParam) && $AddDeletedParam != '0') {
+            throw new SugarApiExceptionMissingParameter('Missing required parameter: add_deleted');
+        }
+
+        $isTrueErrors = $this->validator->validate($AddDeletedParam, new Assert\IsTrue());
```

Comment:
I believe you are trying to use `Assert\IsTrue()` and `Assert\IsFalse()` to asset that this is a boolean string value of "1" or "0" using input validation framework. It seems a little roundabout.

Or if the input is expected to be passed as a String using something like `Assert\Type("string")` and then check if it equals 1 or 0, true or false, etc.

Or you could try using an Expression constraint with something like `"value in ['1', '0']"`
https://symfony.com/doc/3.4/reference/constraints/Expression.html